app/parse.py

```diff
@@ -15,8 +24,78 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product_soup: Tag) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description"
+        ).text.replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".ws-icon.ws-icon-star")),
+        num_of_reviews=int(
+            product_soup.select_one("p.pull-right").text.split()[0]
+        )
+    )
+
+
+def get_product_blocks(page_soup: BeautifulSoup) -> list[Tag]:
+    blocks = page_soup.select(".thumbnail")
+    return [block for block in blocks]
+
+
+def parse_single_page(url: str, driver: webdriver) -> list[Product]:
+    driver.get(url)
+
+    try:
+        more_button = driver.find_element(
+            By.CSS_SELECTOR,
+            "a.btn.btn-primary.btn-lg.btn-block.ecomerce-items-scroll-more"
+        )
+    except NoSuchElementException:
+        more_button = None
+
+    if more_button is not None:
+        while True:
```

Comment:
You can simplify it using <button_object>.is_displayed() and <button_object>.click() methods here